src/screens/Testing/TestProcess/index.tsx

```diff
@@ -8,23 +8,162 @@ import {AppButton} from '@src/components/ui/AppButton';
 import {ViewFlexCenter, CountQuestionBox, ViewBlock} from './styles';
 import {ProgressBar} from '@src/components/ProgressBar';
 import {ProgressType} from '@src/components/ProgressBar/ProgressView';
-import {useAppSelector} from '@hooks/hooks';
+import {useAppDispatch, useAppNavigate, useAppSelector} from '@hooks/hooks';
+import {ScreenList} from '@src/navigation/navigation';
+import {clearStateResult, setStateResult} from '@src/bll/resultReducer';
+
+export type ResultType = {
+  id: number;
+  questionStatus: 'default' | 'active' | 'right' | 'error';
+  answer: string;
+};
 
 export const TestProcess = () => {
+  const {navigate} = useAppNavigate();
+  const dispatch = useAppDispatch();
+  const resultData = useAppSelector(state => state.resultReducer.result);
+  const checkData = useAppSelector(state => state.checkReducer.options);
   const quizIdMocState = useAppSelector(state => state.processReducer);
   const [numAnswer, setNumAnswer] = useState<number>(1);
+  const [singleAnswer, setSingleAnswer] = useState<string[]>([]);
+  const [isActiveRadio, setIsActiveRadio] = useState<number | undefined>(undefined);
+  const checkDataAnswers = checkData.filter(e => e.check);
+  const checkAnswer = checkDataAnswers.map(e => e.label);
```

Comment:
create util function for this. `getCheckedAnswers(checkedData)`

Reply:
and you can combine logic `filter` and `map` methods into `reduce` method:
`checkedData.reduce((acc, el)=> {
    if(el.check) acc.push(el.label)
    return acc;
  }, [])`